hyperparameter_tuning/huggingface_multiclass_text_classification_20_newsgroups/hpo_huggingface_text_classification_20_newsgroups.ipynb

```diff
@@ -0,0 +1,1363 @@
+    "    },\n",
+    "    inplace=True,\n",
+    ")\n",
+    "\n",
+    "# replace to computer\n",
+    "all_categories_df[\"label\"].replace(\n",
+    "    {\n",
+    "        \"comp.windows.x\": \"computer\",\n",
+    "        \"comp.sys.ibm.pc.hardware\": \"computer\",\n",
+    "        \"comp.os.ms-windows.misc\": \"computer\",\n",
+    "        \"comp.graphics\": \"computer\",\n",
+    "        \"comp.sys.mac.hardware\": \"computer\",\n",
+    "    },\n",
+    "    inplace=True,\n",
+    ")\n",
+    "# replace to sales\n",
+    "all_categories_df[\"label\"].replace({\"misc.forsale\": \"sales\"}, inplace=True)\n",
+    "\n",
+    "# replace to science\n",
+    "all_categories_df[\"label\"].replace(\n",
+    "    {\n",
+    "        \"sci.crypt\": \"science\",\n",
+    "        \"sci.electronics\": \"science\",\n",
+    "        \"sci.med\": \"science\",\n",
+    "        \"sci.space\": \"science\",\n",
+    "    },\n",
+    "    inplace=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we are left with 6 categories, which is much better."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "all_categories_df[\"label\"].value_counts()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's calculate number of words for each row."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "all_categories_df[\"word_count\"] = all_categories_df[\"text\"].apply(lambda x: len(str(x).split()))\n",
+    "all_categories_df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's get basic statistics about the dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "all_categories_df[\"word_count\"].describe()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can see that the mean value is around 159 words. However, there are outliers, such as a text with 11351 words. This can make it harder for the model to result in good performance. We will take care to drop those rows."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's drop empty rows first."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "no_text = all_categories_df[all_categories_df[\"word_count\"] == 0]\n",
+    "print(len(no_text))\n",
+    "\n",
+    "# drop these rows\n",
+    "all_categories_df.drop(no_text.index, inplace=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's drop the rows that are longer than 256 words. This is done to make it easy for the model to train without outliers."
+   ]
```

Comment:
Can you provide some insight here into why 256 was chosen as a threshold?

Reply:
Added more details about why this value was chosen.